src/Corax/Pipeline/ITransformer.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Corax.Pipeline
+{
+    public interface ITransformer
+    {
+        bool RequiresTokenSpace => false;
+        bool RequiresBufferSpace => false;
+
+        /// <summary>
+        /// The token space multiplier ensures that the caller would call the Transform method providing enough buffer space to avoid an exception. 
+        /// </summary>
+        float TokenSpaceMultiplier => 1;
```

Comment:
why float ?

Reply:
Because the idea is that we are allowing it to shrink too.